packages/flutter/lib/src/material/app_bar_theme.dart

```diff
@@ -243,10 +344,194 @@ class AppBarTheme with Diagnosticable {
     );
   }
 
+  @override
+  bool updateShouldNotify(covariant AppBarTheme oldWidget) => data != oldWidget.data;
+
+  @override
+  Widget wrap(BuildContext context, Widget child) {
+    return AppBarTheme(data: data, child: child);
+  }
+}
+
+/// Defines default property values for descendant [AppBar] widgets.
+///
+/// Descendant widgets obtain the current [AppBarThemeData] object using
+/// [AppBarTheme.of]. Instances of [AppBarThemeData] can be
+/// customized with [AppBarThemeData.copyWith].
+///
+/// Typically an [AppBarThemeData] is specified as part of the overall [Theme]
+/// with [ThemeData.appBarTheme].
+///
+/// All [AppBarThemeData] properties are `null` by default. When null, the [AppBar]
+/// will use the values from [ThemeData] if they exist, otherwise it will
+/// provide its own defaults. See the individual [AppBar] properties for details.
+///
+/// See also:
+///
+///  * [AppBar], which is the widget that this theme configures.
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class AppBarThemeData with Diagnosticable {
+  /// Creates an app bar theme that can be used with [ThemeData.appBarTheme].
+  const AppBarThemeData({
+    this.backgroundColor,
+    this.foregroundColor,
+    this.color,
+    this.elevation,
+    this.scrolledUnderElevation,
+    this.shadowColor,
+    this.surfaceTintColor,
+    this.shape,
+    this.iconTheme,
+    this.actionsIconTheme,
+    this.centerTitle,
+    this.titleSpacing,
+    this.leadingWidth,
+    this.toolbarHeight,
+    this.toolbarTextStyle,
+    this.titleTextStyle,
+    this.systemOverlayStyle,
+    this.actionsPadding,
+  }) : assert(
+         color == null || backgroundColor == null,
+         'The color and backgroundColor parameters mean the same thing. Only specify one.',
+       );
+
+  /// Overrides the default value of [AppBar.backgroundColor].
+  final Color? backgroundColor;
+
+  /// Overrides the default value of [AppBar.foregroundColor].
+  final Color? foregroundColor;
+
+  /// This property is equivalent to [backgroundColor].
+  /// This is obsolete and will be deprecated in a future release:
+  /// please try to use the [backgroundColor] instead.
+  final Color? color;
```

Comment:
Same for `AppBarTHemeData`, the color should only be a parameter in the constructor and should be deprecated later:)